pkg/workceptor/kubernetes_test.go

```diff
@@ -3950,118 +3951,159 @@ func TestKubeWorkerCfg_Run(t *testing.T) {
-
-		err = os.MkdirAll(testUnitDir, 0o700)
-		if err != nil {
-			t.Logf("Failed to create unit dir for %s: %v", testUnitDir, err)
-		}
-		kubeConfig := workceptor.KubeWorkerCfg{
-			AuthMethod:   "incluster",
-			StreamMethod: "logger",
-		}
+			ctx := context.Background()
+			w, err := workceptor.New(ctx, mockNetceptor, "/tmp")
+			if err != nil {
+				t.Fatalf("Error creating Workceptor: %v", err)
+			}
 
-		kubeUnit := kubeConfig.NewkubeWorker(mockBaseWorkUnit, w, "", "", mockKubeAPI).(*workceptor.KubeUnit)
+			statusLock := &sync.RWMutex{}
+			statusData := &workceptor.StatusFileData{State: 1, ExtraData: &workceptor.KubeExtraData{}}
+			statusCopy := workceptor.StatusFileData{ExtraData: &workceptor.KubeExtraData{
+				KubeNamespace: testNamespace,
+				PodName:       tc.podName,
+			}}
+
+			mockBaseWorkUnit.EXPECT().GetStatusLock().Return(statusLock).AnyTimes()
+			mockBaseWorkUnit.EXPECT().GetStatusWithoutExtraData().Return(statusData).AnyTimes()
+			mockBaseWorkUnit.EXPECT().GetStatusCopy().Return(statusCopy).AnyTimes()
+			mockBaseWorkUnit.EXPECT().GetContext().Return(context.Background()).AnyTimes()
+			mockBaseWorkUnit.EXPECT().UnitDir().Return(testUnitDir)
+			mockBaseWorkUnit.EXPECT().GetWorkceptor().Return(w).AnyTimes()
+			mockBaseWorkUnit.EXPECT().UpdateBasicStatus(workceptor.WorkStateRunning, gomock.Any(), gomock.Any())
+			mockBaseWorkUnit.EXPECT().Init(w, "", "", workceptor.FileSystem{})
+			mockBaseWorkUnit.EXPECT().UpdateBasicStatus(workceptor.WorkStateSucceeded, gomock.Any(), gomock.Any())
 
-		// Create a pod with container that has terminated with exit code 1
-		existingPod := &corev1.Pod{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      testPodName,
-				Namespace: testNamespace,
-			},
-			Status: corev1.PodStatus{
-				Phase: corev1.PodRunning,
-				Conditions: []corev1.PodCondition{
-					{Type: corev1.PodReady, Status: corev1.ConditionTrue},
+			err = os.MkdirAll(testUnitDir, 0o700)
+			if err != nil {
+				t.Logf("Failed to create unit dir for %s: %v", testUnitDir, err)
+			}
+			kubeConfig := workceptor.KubeWorkerCfg{
+				AuthMethod:   "incluster",
+				StreamMethod: "logger",
+			}
+
+			kubeUnit := kubeConfig.NewkubeWorker(mockBaseWorkUnit, w, "", "", mockKubeAPI).(*workceptor.KubeUnit)
+
+			// Create a pod with container that has terminated with exit code 1
+			existingPod := &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      tc.podName,
+					Namespace: testNamespace,
 				},
-				ContainerStatuses: []corev1.ContainerStatus{
-					{
-						Name: workceptor.WorkerContainerName,
-						State: corev1.ContainerState{
-							Terminated: &corev1.ContainerStateTerminated{
-								ExitCode: 1,
-								Reason:   "Error",
-								Message:  "Process completed with errors",
+				Status: corev1.PodStatus{
+					Phase: corev1.PodRunning,
+					Conditions: []corev1.PodCondition{
+						{Type: corev1.PodReady, Status: corev1.ConditionTrue},
```

Comment:
similar note here - Once everything is done I think the ready condition is False. But probably not worth the cycles to change this 😁